src/test/java/com/github/studym/studymarathon/Board/BoardRepositoryTests.java

```diff
@@ -0,0 +1,71 @@
+package com.github.studym.studymarathon.Board;
+
+import com.github.studym.studymarathon.domain.board.entity.Board;
+import com.github.studym.studymarathon.domain.board.repository.BoardRepository;
+import com.github.studym.studymarathon.domain.member.entity.Member;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+@SpringBootTest
+public class BoardRepositoryTests {
+
+    @Autowired
+    private BoardRepository boardRepository;
+
+
+    @Test
+    public void insertDummies() {
+        IntStream.rangeClosed(1, 199).forEach(i -> {
+            Board board = Board.builder()
+                    .title("제목" + i)
+                    .content("내용애용내용" + i)
+                    .author("작성자" + i)
+                    .build();
+            System.out.println(boardRepository.save(board));
+        });
+    }
+
+    @Test
+    public void updateTest() {
+        Optional<Board> result = boardRepository.findById(298L);
+
+        if (result.isPresent()) {
+            Board board = result.get();
+
+            board.changeTitle("변경한 제목");
+            board.changeContent("변경한 내용");
+
+            boardRepository.save(board);
+        }
+    }
+
+    @Test
+    public void test(){
+        logic(199L);
+    }
+
+    public Board updateOrElse(Long id){
+        Optional<Board> optional = boardRepository.findById(id);
+
+        return optional.orElse(null);
+    }
+
+
+    public void logic(Long id){
+        Board board = updateOrElse(id);
+        board.changeContent("바꾼내용이라구우");
+        System.out.println(boardRepository.save(board));
+    }
```

Comment:
조회된 결과가 없을 경우엔 null을 출력하고 싶은 것이 아닌 이상, 아래와 같이 ifPresent를 사용하여 조회된 결과가 있을 경우에만 로직을 실행하도록 하는 것이 좋아보임

```suggestion
    @Test
    public void updateIfPresentTest() {
            boardRepository.findById(199L)
                .ifPresent(board -> {
                    board.changeContent("바꾼내용이라구우");
                    System.out.println(boardRepository.save(board));
                });
    }
```

혹은 아래처럼 조회된 결과가 없을 경우 별도의 처리를 할 수 있는 ifPresentOrElse를 사용해도 괜찮아보임

```suggestion
    @Test
    public void updateIfPresentTest() {
        boardRepository.findById(199L)
                .ifPresentOrElse(board -> {
                    board.changeContent("바꾼내용이라구우");
                    System.out.println(boardRepository.save(board));
                }, () -> System.out.println("조회된 결과 없음"));
    }
```

이거랑은 별개로 standard output을 사용한 테스팅은 지양하는 것이 좋음

Reply:
그리고 jpa 사용 시 트랜잭션 내에서 조회한 entity에 대해서는 자동으로 dirty checking을 하기 때문에 굳이 코드상에서 명시적으로 save를 해주지 않아도 update 쿼리를 날리게 되어있음
[Dirty Checking이란?](https://jojoldu.tistory.com/415)
추후 로직 구현 시 참고하면 좋을듯

Reply:
> 그리고 jpa 사용 시 트랜잭션 내에서 조회한 entity에 대해서는 자동으로 dirty checking을 하기 때문에 굳이 코드상에서 명시적으로 save를 해주지 않아도 update 쿼리를 날리게 되어있음 [Dirty Checking이란?](https://jojoldu.tistory.com/415) 추후 로직 구현 시 참고하면 좋을듯

이쪽도 지식이 너무 부족했네 도움 많이 된거 같음 고맙다